src/api/app/controllers/webui/staging/workflows_controller.rb

```diff
@@ -1,13 +1,16 @@
 class Webui::Staging::WorkflowsController < Webui::WebuiController
   VALID_STATES_WITH_REQUESTS = [:acceptable, :accepting, :review, :testing, :building, :failed, :unacceptable].freeze
 
-  before_action :require_login, except: [:show]
+  # TODO: Remove this when we'll refactor kerberos_auth
+  before_action :kerberos_auth, except: [:show]
   before_action :set_project, only: [:new, :create]
   before_action :set_workflow_project, except: [:new, :create]
   before_action :set_staging_workflow, except: [:new, :create]
-  after_action :verify_authorized, except: [:show, :new]
+  after_action :verify_authorized, except: [:show]
 
   def new
+    authorize ::Staging::Workflow, :new?
```

Comment:
If we have this line right under `@staging_workflow = @project.build_staging` (on line 19 right now), we could check if the user can instantiate the new staging workflow.

Reply:
Thank you for the suggestion!